### DISCUSSION TITLE
Camera or scene masks in upbge

### DISCUSSION THREAD

**Discussion by HolandonHdunto** [UPBGE/upbge#1668] [Ideas/Features for UPBGE 0.2.5+]
It would be nice if there was a way to put masks on cameras or scenes in upbge, where all objects outside the mask would be hidden, areas of the object outside the mask would be hidden and the collision would only work inside the mask. Almost an iframe.

Here are some representative images.

![Mascara](https://user-images.githubusercontent.com/89164699/155593696-f559685a-5b63-48b0-a862-3e37463ac440.png)

Here is a representative image showing before.

![Cena](https://user-images.githubusercontent.com/89164699/155593961-759fd0c5-8eff-48e9-a58e-d4ffcec65b76.png)

Here is a representative image showing later what it would look like.

![ControleCamera](https://user-images.githubusercontent.com/89164699/155594181-3d49c456-93d3-476d-b3cc-279baed165d1.png)

![CenaComMascara](https://user-images.githubusercontent.com/89164699/155594228-15753432-6f6d-42c6-9e16-64ea95daa2e2.png)

Here's a video trying to produce the same effect.

https://www.youtube.com/watch?v=-B0vvTrXI2Q

**Comment by rpaladin** [comment]
People have already showcased & shared how create such a mechanic in UPBGE with material nodes:

https://www.youtube.com/watch?v=4Seqm826oWk

**Reply by HolandonHdunto** [reply]
Well, the mouse can access the area of ​​the object outside the mask, but after a while after launching this topic I ended up thinking of a more practical and simple way to make the mask.

**Comment by rpaladin** [comment]
You might also check this other UPBGE user's work. He was able to produce realistic masking using material shaders:

https://github.com/KorolevAndrey/UPBGE-BGE-Outline/tree/main/Silhouette%20Outline%20POST%20Effect

**Reply by HolandonHdunto** [reply]
Being able to make the mask in a practical way, but the Custom Viwport was complicated because we don't know exactly where it will be positioned with this configuration, it would be simpler to have the x and y position, and the scale of the scene. Another thing is the texts, the materials don't work in the texts, and you can't create an Input Text for example.
https://youtu.be/BJ-97nQrd3k
